packages/appium/lib/appium.js

```diff
@@ -276,34 +275,27 @@ class AppiumDriver extends BaseDriver {
   }
 
   attachUnexpectedShutdownHandler (driver, innerSessionId) {
-    const removeSessionFromMasterList = (cause = new Error('Unknown error')) => {
-      log.warn(`Closing session, cause was '${cause.message}'`);
+    const onShutdown = (cause = new Error('Unknown error')) => {
+      log.warn(`Ending session, cause was '${cause.message}'`);
+
+      if (this.sessionPlugins[innerSessionId]) {
+        for (const plugin of this.sessionPlugins[innerSessionId]) {
+          if (_.isFunction(plugin.onUnexpectedShutdown)) {
+            log.debug(`Plugin ${plugin.name} defines an unexpected shutdown handler; calling it now`);
+            plugin.onUnexpectedShutdown(driver, cause);
```

Comment:
should it be awaited?

Reply:
i thought about this, but that would break the signature of this method and require us to put `awaits` in the other drivers. better to require the plugin author to use appropriate error handling if they wish it to be an `async` function. we are any rate not going to do anything if there is an error here.

Reply:
putting await won't break anything. It's simply ignored if the returned result is not a promise